http/vibe/http/server.d

```diff
@@ -765,12 +765,21 @@ final class HTTPServerRequest : HTTPRequest {
 
 		/** Contains the parsed Json for a JSON request.
 
-			Remarks:
-				This field is only set if HTTPServerOption.parseJsonBody is set.
-
-				A JSON request must have the Content-Type "application/json" or "application/vnd.api+json".
+			A JSON request must have the Content-Type "application/json" or "application/vnd.api+json".
 		*/
-		Json json;
+		@property ref const(Json) json() {
+			if (_json.isNull) {
+				if (icmp2(contentType, "application/json") == 0 || icmp2(contentType, "application/vnd.api+json") == 0 ) {
+					auto bodyStr = () @trusted { return cast(string) bodyReader.readAll(); } ();
+					if (!bodyStr.empty) _json = parseJson(bodyStr);
+				} else {
+					_json = Json.undefined;
```

Comment:
This is done to emulate the current behavior when no json content type is present.